### ISSUE TITLE
name 'socket' is not defined in email.utils.make_msgid()

### ISSUE THREAD

**Issue by jshenguru** [Nuitka/Nuitka#2293]
**Problem**: Exe complains about 'socket' not defined

Nuitka version:
```
1.6.3
Commercial: 2.1.1
Python: 3.9.15+ (heads/3.9-dirty:5f7d77a601, May 18 2023, 09:54:58) [MSC v.1929 64 bit (AMD64)]
Flavor: Nuitka Python
Executable: C:\Users\test\Documents\Python3.9-amd64\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 10
Version C compiler: C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\Hostx64\x64\cl.exe (cl 14.2).

```
Options:
```
--show-scons\
        --standalone \
        --remove-output \
        --company-name=$(SOFTWARE_COMPANY_NAME) \
        --product-name=$(PRODUCT_NAME) \
        --file-version=$(FILE_VERSION) \
        --product-version=$(PRODUCT_VERSION) \
        --file-description=$(FILE_DESCRIPTION) \
        --copyright=$(COPYRIGHT) \
        --trademarks=$(TRADEMARK) \
        --windows-icon-from-ico=$(ICON_FILE) \
        --embed-data-files-run-time-pattern=* \
```
Minimal example:

`import email.utils
email.utils.make_msgid()
`
Error:
```
File "C:\MyAPp\bin\email\utils", line 192, in make_msgid
NameError: name 'socket' is not defined
```

**Comment by kayhayen** [comment]
Ah, that's a similar one, I will fix it right away I think.

**Comment by kayhayen** [comment]
So, this was done:

```
- module-name: 'email.utils'
  anti-bloat:
    - description: 'avoid socket module usage'
      replacements_plain:
        'import socket': ''
      change_function:
        'make_msgid': 'un-callable'
      when: 'not use_socket'
```

**Comment by kayhayen** [comment]
So on the staging branch, I added the missing ability. If you do `--noinclude-custom-mode=socket:allow` the `use_socket` becomes true, disabling above rule. It's also on factory, I will be looking at including it in the hotfix as well, but anti-bloat plugin got refactored quite a bit, but I think it should be portable.

**Comment by kayhayen** [comment]
So, that's also on the hotfix branch, 1.6.4, so far, there is no automatic commercial branches from these, of which your report reminds me. It's not trivial to make though.

**Comment by jshenguru** [comment]
> So, that's also on the hotfix branch, 1.6.4, so far, there is no automatic commercial branches from these, of which your report reminds me. It's not trivial to make though.

Thank you very much for the quick fix on this.

Just so I understand correctly, if I do the following to my 1.6.3 commercial, this issue should be gone, right?

1. Add the following after line 141 in the `Lib\site-packages\nuitka\plugins\standard\ANtiBloatPlugin.py`
```
            if mode == "allow":
                self.control_tags["use_%s" % module_name] = True
```
2. While compiling my project, add options of `--noinclude-custom-mode=socket:allow`

**Comment by kayhayen** [comment]
Yes, that should work. It's also a commit on the hotfix branch, that you can cherry-pick too, but I believe it's going to be out any day now anyway as a 1.6.4 in the offical branches too.

**Comment by jshenguru** [comment]
> Yes, that should work. It's also a commit on the hotfix branch, that you can cherry-pick too, but I believe it's going to be out any day now anyway as a 1.6.4 in the offical branches too.

Thank you very much!

**Comment by kayhayen** [comment]
Part of 1.6.4 release, although I feel like this one deserves a different treatment, where rather than being removed, this introduces a special sort of module, that when used raises a `RuntimeError`, but this is maybe for later. I will add the stream tag to remind me od that. The handling should only be done, if `socket` is not used by other modules.

**Comment by jshenguru** [comment]
> Part of 1.6.4 release, although I feel like this one deserves a different treatment, where rather than being removed, this introduces a special sort of module, that when used raises a `RuntimeError`, but this is maybe for later. I will add the stream tag to remind me od that. The handling should only be done, if `socket` is not used by other modules.

I will upgrade my Nuitka to 1.6.4 today. I still need to use the option `add options of --noinclude-custom-mode=socket:allow` for compilation, right?

**Comment by kayhayen** [comment]
Yes, the changes I intend will add something, that will inform you of this at runtime, but without removing this one, every program needs socket and its extension module when in fact, it does not really do it. The option will be needed in this form or another form in the future still.
